src/huggingface_hub/hf_api.py

```diff
@@ -553,19 +552,21 @@ def _is_valid_token(self, token: str):
         except HTTPError:
             return False
 
-    def _validate_or_retrieve_token(self, token: Optional[Union[str, bool]] = None):
+    def _validate_or_retrieve_token(
+        self,
+        token: Optional[str] = None,
+        name: Optional[str] = None,
+        function_name: Optional[str] = None,
+    ):
         """
-        Either retrieves stored token or validates passed token.
-
+        Retrieves and validates stored token or validates passed token.
         Args:
-            token (`str`, *optional*):
-                The token to check for validity
-
-        Returns:
-            `str`: The valid token
-
-        Raises:
-            `ValueError`: if the token is invalid.
+            token (``str``, `optional`):
+                Hugging Face token. Will default to the locally saved token if not provided.
+            name (``str``, `optional`):
+                Name of the repository.
+            function_name (``str``, `optional`):
+                If called from a function, name of that function for deprecation warning.
```

Comment:
Missing `Returns` and `Raises` sections.

It would also be nice to explain why `name` and `function_name` are required and that they'd be removed in a future version (in v0.7 maybe?)

Reply:
Done!